src/github.com/matrix-org/dendrite/syncapi/consumers/roomserver.go

```diff
@@ -177,6 +202,44 @@ func (s *OutputRoomEvent) lookupStateEvents(
 	return result, nil
 }
 
+func (s *OutputRoomEvent) updateMemberEvent(
+	event gomatrixserverlib.Event, keyID gomatrixserverlib.KeyID,
+	privateKey []byte,
+) (gomatrixserverlib.Event, error) {
+	if event.Type() != "m.room.member" {
+		return event, nil
+	}
+	membership, err := event.Membership()
+	if err != nil {
+		return event, err
+	}
+	if membership != "join" {
+		return event, nil
+	}
```

Comment:
Ah, there would be an issue with that. The reason I use the `event.MemberContent` here: https://github.com/matrix-org/dendrite/blob/37a9db5e6ac4d9b8fa7e9457eeceb4903e794fd8/src/github.com/matrix-org/dendrite/syncapi/consumers/roomserver.go#L45 is that, because `event.Content()` returns `[]byte`, when I directly set it in "prev_content" without unmarshalling it, it will be processed as a string ("prev_content" will just be an useless string in the actual request).
To fix this, I'm unmarshalling it into a structure specific to the event. So, unless we have a structure for the content of each event type the roomserver will receive, and a way to automatically use the right one depending on the type, it won't work :/